File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max_prices = 0
        max_profits = 0
        for i in range(len(prices)-1, -1, -1):
            max_prices = max(max_prices, prices[i])
            max_profits = max(max_profits, max_prices - prices[i])
        return max_profits

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int result = 0, min = prices[0];
        for(int i = 1; i < prices.size(); i++){
            if(prices[i] < min){
                min = prices[i];
            }
            if(prices[i] > min){
                if(prices[i] - min > result){
                    result = prices[i] - min;
                }
            }
        }
        return result;
    }
};